pilot/cmd/pilot-discovery/admission/controller.go

```diff
@@ -0,0 +1,31 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
```

Comment:
A line or two explaining what this does.

Reply:
Should this just be added to the `NewController` method?  What additional explanation are you looking for exactly? Are you asking for an explanation of what an admission controller is?

Reply:
Yep. Something to the effect of - this code watches the API server for new CRDs and validates them based on various ...

Reply:
This looks like a fairly small function. Can you just move it to `admit` package?